syntax = "proto3";

package proto_transport_db;

import "map_renderer.proto";
import "transport_router.proto";

message Stop {
    string name = 1;
    double latitude = 2;
    double longitude = 3;
}

message Bus {
    string name = 1;
    repeated string stops = 2;
    bool is_circle = 3;
    string second_final_stop = 4;  
}

message Distances {
    string from = 1;
    string to = 2;
    int32 distance = 3;
}

message TransportCatalogue {
    repeated Bus buses = 1;
    repeated Stop stops = 2;
    repeated Distances distance = 3;
    proto_map.RenderSettings render_settings = 4;
    Router router = 5;
}

